src/connection/manager.js

```diff
@@ -185,24 +199,24 @@ class ConnectionManager {
             })
             this.switch.connection.add(connection)
 
-            if (peerInfo.multiaddrs.size > 0) {
-              // with incomming conn and through identify, going to pick one
-              // of the available multiaddrs from the other peer as the one
-              // I'm connected to as we really can't be sure at the moment
-              // TODO add this consideration to the connection abstraction!
-              peerInfo.connect(peerInfo.multiaddrs.toArray()[0])
-            } else {
-              // for the case of websockets in the browser, where peers have
-              // no addr, use just their IPFS id
-              peerInfo.connect(`/ipfs/${b58Str}`)
+            // Only update if it's not already connected
+            if (!peerInfo.isConnected()) {
```

Comment:
How reliable is peerInfo.isConnected()? I ask because it seems like things are put in the peer book from a lot of different places in the code

Reply:
`PeerBook.put()` is definitely overused. I think clearly defining the peer discovery pipeline and responsibility around who actually should be putting would be helpful.  `PeerInfo.isConnected` has been showing questionable reliability in my local checks, but I haven't determine the root cause of that yet. I did a search of modules in js-ipfs for `.connect` calls (although I need to check `.disconnect`) to see if any modules are calling those, which they shouldn't be.

If `PeerInfo`returns a connected address we should be confident in its validity though. The main goal here is to avoid stomping over a more useful dialed address with an incoming address, as we know for a fact the former was used in a successful dial.